core/src/main/java/io/aiven/kafka/tieredstorage/metadata/SegmentCustomMetadataField.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.metadata;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+import org.apache.kafka.common.protocol.types.Field;
+import org.apache.kafka.common.protocol.types.Field.TaggedFieldsSection;
+import org.apache.kafka.common.protocol.types.Schema;
+import org.apache.kafka.common.protocol.types.Type;
+
+import io.aiven.kafka.tieredstorage.ObjectKey;
+
+// ordinal index define values on custom metadata fields and cannot be changed without breaking compatibility.
+public enum SegmentCustomMetadataField {
+    TOTAL_SIZE(new Field("total_size", Type.VARLONG), SegmentCustomMetadataBuilder::totalSize),
+    OBJECT_PREFIX(new Field("object_prefix", Type.COMPACT_STRING), b -> b.objectKey.prefix()),
+    OBJECT_KEY(new Field("object_key", Type.STRING), b -> ObjectKey.mainPath(b.segmentMetadata));
```

Comment:
Any reason not to use `COMPACT_STRING` here as well?

Reply:
Not really, forgot to align these. Should be fixed now 👍🏽